flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecHashJoin.java

```diff
@@ -293,4 +299,61 @@ private long getLargeManagedMemory(FlinkJoinType joinType, ExecNodeConfig config
         // large one
         return Math.max(hashJoinManagedMemory, sortMergeJoinManagedMemory);
     }
+
+    @Override
+    public boolean supportFusionCodegen() {
+        RowType leftType = (RowType) getInputEdges().get(0).getOutputType();
+        LogicalType[] keyFieldTypes =
+                IntStream.of(joinSpec.getLeftKeys())
+                        .mapToObj(leftType::getTypeAt)
+                        .toArray(LogicalType[]::new);
+        RowType keyType = RowType.of(keyFieldTypes);
+        FlinkJoinType joinType = joinSpec.getJoinType();
+        HashJoinType hashJoinType =
+                HashJoinType.of(
+                        leftIsBuild,
+                        joinType.isLeftOuter(),
+                        joinType.isRightOuter(),
+                        joinType == FlinkJoinType.SEMI,
+                        joinType == FlinkJoinType.ANTI);
+        // TODO decimal and multiKeys support and all HashJoinType support.
+        return LongHashJoinGenerator.support(hashJoinType, keyType, joinSpec.getFilterNulls());
```

Comment:
Currently, we only support codegen when the join key is a single key and is a long type, otherwise, it is all hardcode, so the first version we currently follow this behavior. For multi-key scenarios, we need another PR to complete them, which are two separate work.